### PR TITLE
Use new fixtures in backend tests

### PR DESCRIPTION
**Pull request by lukoktonos**
PR 1 of 2 into feature branch **cqm-patient-tests** for switching test fixtures to use cqm_patient fixtures.  This one updates all fixtures and makes changes so that the cqm_patient fixtures are used and loaded into the backend tests.  Most of the altered tests fail due to differences between Record and CQM::Patient.  Fixing those tests will be part of a future task.

I also added a few `TODO(cqm_patient)` items to guide whoever does the task of fixing the tests themselves where I noticed things that need to be done.

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1933
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [x] Tests are included and test edge cases (N/A editing tests)
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable) (N/A editing tests)
- [x] Test fixtures updated and documented as necessary ( see [internal wiki](https://gitlab.mitre.org/bonnie/internal-documentation/wikis/testing#test-fixtures) ) (This is still a TODO and makes sense to do after this is reviewed rather than before)
- [x] Code coverage has not gone down and all code touched or added is covered. 
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: 
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 
- [x] Automated regression test(s) pass N/A

**Reviewer 1:**

Name: @hackrm 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases (N/A - Verified that all frontend tests passed & the 'export_fixtures_test.rb' tests passed. All other backend tests that are currently failing will be addressed in a following PR)
- [x] You have tried to break the code

**Reviewer 2:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: test/unit/patient_builder_test.rb [hunk 1]
```diff
@@ -4,29 +4,33 @@ class PatientBuilderTest < ActiveSupport::TestCase
...
-    @p1_lab_test_1xx_sdc = @p1.source_data_criteria.find { |dc| dc['description'] == 'Laboratory Test, Performed: Urine Protein Tests' && dc['start_date'] == 1343635200000 }
-
-    @p2 = Record.find_by(last: 'Denex', first: 'Fail_Hospice_Not_Performed')
-    vs_oids = @p2.source_data_criteria.collect { |dc| Measures::PatientBuilder.get_vs_oids(dc) }.flatten.uniq
+    @p1_diagnosis_diabetes_sdc = @p1.qdmPatient.dataElements.find_by( description: 'Diagnosis: Diabetes' )
+    @p1_wellness_visit_sdc = @p1.qdmPatient.dataElements.find_by( description: 'Encounter, Performed: Annual Wellness Visit' )
+    # TODO(cqm_patient): need to update queries below as start_date is no longer relevant
+    @p1_lab_test_kidney_sdc = @p1.qdmPatient.dataElements.find_by( description: 'Laboratory Test, Performed: Urine Protein Tests', start_date: 1343634300000 )
+    @p1_lab_test_1xx_sdc = @p1.qdmPatient.dataElements.find_By( description: 'Laboratory Test, Performed: Urine Protein Tests', start_date: 1343635200000 )
+    binding.pry
```
Comment (hackrm) [inline test/unit/patient_builder_test.rb]: There is a debugging statement here still.

File: test/rake/bonnie_patients_test.rb [hunk 2]
```diff
@@ -217,23 +217,23 @@ class BonniePatientsTest < ActiveSupport::TestCase
...
+  test 'move_patients_csv with bad information' do
     ENV['CSV_PATH'] = File.join(Rails.root, 'test', 'fixtures', 'csv', 'bad_transfers.csv')
 
 
-    source_patients = Record.where(measure_ids:@source_hqmf_set_id)
-    dest_patients = Record.where(measure_ids:@dest2_hqmf_set_id)
-    user_patients = Record.where(user_id:@source_user.id)
+    source_patients = CQM::Patient.where(measures: CQM::Measure.find(hqmf_set_id: @source_hqmf_set_id))
+    dest_patients = CQM::Patient.where(measures: CQM::Measure.find(hqmf_set_id: @dest2_hqmf_set_id))
+    user_patients = CQM::Patient.tient.where(user_id:@source_user.id)
```
Comment (hackrm) [inline test/rake/bonnie_patients_test.rb]: It looks like the '.tient' was erroneously added here.

File: test/rake/bonnie_patients_test.rb [hunk 3]
```diff
@@ -252,29 +252,29 @@ class BonniePatientsTest < ActiveSupport::TestCase
                   "\e[#{31}m#{"[Error]"}\e[0m\t\tunable to move records in bonnie@example.com from test3:Median Time from ED Arrival to ED Departure for Discharged ED Patients to CMS160v6:Depression Utilization of the PHQ-9 Tool\n\n"
                  ) { Rake::Task['bonnie:patients:move_patients_csv'].execute }
 
-    source_patients = Record.where(measure_ids:@source_hqmf_set_id)
-    dest_patients = Record.where(measure_ids:@dest2_hqmf_set_id)
-    user_patients = Record.where(user_id:@source_user.id)
+    source_patients = CQM::Patient.where(measures: CQM::Measure.find(hqmf_set_id: @source_hqmf_set_id))
+    dest_patients = CQM::Patient.tient.where(measures: CQM::Measure.find(hqmf_set_id: @dest2_hqmf_set_id))
```
Comment (hackrm) [inline test/rake/bonnie_patients_test.rb]: It looks like the '.tient' was erroneously added here.

**Review by hackrm — APPROVED** [review]

File: test/mailers/user_mailer_test.rb [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 class UserMailerTest < ActionMailer::TestCase
   setup do
-    @user = User.new(email: 'email_test@test.com', first: 'first', last: 'last', password: 'Test1234!')
+    @user = User.new(email: 'email_test@test.com', givenNames: 'first', familyName: 'last', password: 'Test1234!')
```
Comment (daco101) [inline test/mailers/user_mailer_test.rb]: I don't believe we are renaming first/last in the user model.

File: test/rake/bonnie_users_test.rb [hunk 5]
```diff
@@ -3,7 +3,7 @@
 class BonnieUsersTest < ActiveSupport::TestCase
   setup do
     dump_database
-    @user = User.new(email: 'test@test.com', first: 'first', last: 'last', password: 'Test1234!')
+    @user = User.new(email: 'test@test.com', givenNames: 'first', familyName: 'last', password: 'Test1234!')
```
Comment (daco101) [inline test/rake/bonnie_users_test.rb]: Same comment about user fields not being renamed.

File: test/rake/cql_test.rb [hunk 6]
```diff
@@ -6,7 +6,7 @@ class CqlTest < ActiveSupport::TestCase
     dump_database
     @cql_mat_export = File.new File.join('test', 'fixtures', 'cql_measure_exports', 'Test134_v5_4_Artifacts.zip')
     @cql_mat_export_drc = File.new File.join('test', 'fixtures', 'cql_measure_exports', 'CMS26v5_Artifacts_direct_reference_code.zip')
-    @user = User.new(email: 'test@test.com', first: 'first' , last: 'last',password: 'Test1234!')
+    @user = User.new(email: 'test@test.com', givenNames: 'first', familyName: 'last', password: 'Test1234!')
```
Comment (daco101) [inline test/rake/cql_test.rb]: And here.

File: test/unit/user_test.rb [hunk 7]
```diff
@@ -5,7 +5,7 @@ class UserTest < ActiveSupport::TestCase
   setup do
     dump_database
     assert_equal 0, User.count
-    @user = User.new(email: "test@test.com", first: "first" , last: 'last',password: 'Test1234!') 
+    @user = User.new(email: "test@test.com", first: 'first', last: 'last', password: 'Test1234!')
```
Comment (daco101) [inline test/unit/user_test.rb:8]: same

File: test/mailers/user_mailer_test.rb (hunk 4 above)
Comment (lukoktonos) [reply test/mailers/user_mailer_test.rb]: Good catch--thought I reverted these already but missed a few

**Review by daco101 — APPROVED** [review]
